docs/articles/documentation/why-testcafe/README.md

```diff
@@ -0,0 +1,126 @@
+
+### Page Proxying: Bridging the Server-Client Gap
+
+You may notice that when you run TestCafe, the URL in the browser’s address bar does not match that of your website. This happens because TestCafe runs an under-the-hood reverse proxy.
+
+The [testcafe-hammerhead](https://github.com/DevExpress/testcafe-hammerhead) proxy intercepts browser requests and adds automation scripts to the requested resources. It then modifies the URLs contained within the resource, so that they point to the proxy. This way, neither the client-side code, nor the resources it communicates with, can tell that the page has been modified. To conceal automation scripts from the rest of the page code, TestCafe mocks the browser API.
+
+The proxying mechanism ensures that the page appears to be hosted at the original URL even to the test code. This is why you can use your website’s actual URL in tests, and pay no mind to the browser’s address bar.
+
+### Browser Sandboxing
+
+At the end of each run, TestCafe deletes all browser cookies, empties the storage, and reloads the page, thereby preventing undesirable interference with subsequent tests. You don’t need to write boilerplate code to reset the app state and reverse the changes your tests make.
+
+Tests that run in parallel operate in independent sandboxed environments. This helps prevent server-side collisions.
+
+### Client-Side Scripts
+
+To perform common testing tasks, TestCafe translates server-side calls to its API into client-side code. However, some scenarios require the execution of client-side code that TestCafe cannot automatically generate. That’s why the framework offers multiple ways to execute user-defined JavaScript.
+
+[Client Scripts](/../guides/advanced-guides/inject-client-scripts.md) inject custom JavaScript files, such as temporary extra dependencies, into the page.
+[Client Functions](../guides/basic-guides/obtain-client-side-info.md) evaluate user-defined JavaScript expressions and pass their return value to the server side. They are useful when you want to examine the page or access the website’s URL.
+The [Selector](../guides/basic-guides/select-page-elements.md) function can launch user-defined client-side code to find a DOM element that cannot be otherwise identified.
+
+However, client-side injections have their limits. Client-side page modifications can disrupt internal TestCafe processes. Client Functions can not return DOM elements. The rule of thumb is to only inject client-side code when you need to pass otherwise unobtainable page data to the server.
```

Comment:
Is this part necessary? This seems too technical for an overview document... Maybe it's best left in the detailed documents linked from the list above. I think we should just remove this paragraph.

Reply:
I'm sort of split on this. I see where you're coming from, but this is important to get across. The current version of the article (the one on the website) describes these features in even more detail. @kirovboris @VasilyStrelyaev

Reply:
I would leave just the fact that they have limitations and need to be used carefully. We can leave the details for the dedicated topic.

In addition, I think this rule of thumb is a bit restricting. We occasionally have to suggest that people use client functions for page modifications (mainly, after they write to support and the developers dig into their case) or to perform some utility operations (like waiting for a custom event/condition in page scripts).

Reply:
```suggestion
> The TestCafe documentation describes the limitations of user-defined client-side scripts.
```

Reply:
ok